airbnb_project/listings/harvester_app/harvester/spiders/airbnb_url_builder.py

```diff
@@ -0,0 +1,215 @@
+        """
+        Construct a full URL using the given parameters.
+
+        Args:
+            param (dict): A dictionary of URL parameters.
+
+        Returns:
+            str: A complete AirBnb search URL.
+        """
+        return f"{AirBnbURLBuilder.SCHEME}://{AirBnbURLBuilder.BASE_URL}/{AirBnbURLBuilder.HOMES_PATH}?{AirBnbURLBuilder.get_params_string(param)}"
+
+    def get_flexible_week(self, *month):
+        """
+        Generate a URL for a flexible one-week stay.
+
+        Args:
+            *month: Variable length argument list of month names. If none provided, uses the current month.
+
+        Returns:
+            str: A URL for a flexible one-week stay search.
+        """
+        if not len(month):
+            month = self._get_current_month()
+        local_param = AirBnbURLBuilder.DEFAULT_PARAMS.copy()
+        local_param['flexible_trip_lengths[]'] = "one_week"
+        local_param['flexible_trip_dates[]'] = month
+        return self._get_full_url(local_param)
+
+    def get_flexible_weekend(self, *month):
+        """
+        Generate a URL for a flexible weekend stay.
+
+        Args:
+            *month: Variable length argument list of month names. If none provided, uses the current month.
+
+        Returns:
+            str: A URL for a flexible weekend stay search.
+        """
+        if not len(month):
+            month = self._get_current_month()
+        local_param = AirBnbURLBuilder.DEFAULT_PARAMS.copy()
+        local_param['flexible_trip_lengths[]'] = "weekend_trip"
+        local_param['flexible_trip_dates[]'] = month
+        return self._get_full_url(local_param)
+
+    def get_flexible_month(self, *month):
+        """
+        Generate a URL for a flexible one-month stay.
+
+        Args:
+            *month: Variable length argument list of month names. If none provided, uses the current month.
+
+        Returns:
+            str: A URL for a flexible one-month stay search.
+        """
+        if not len(month):
+            month = self._get_current_month()
+        local_param = AirBnbURLBuilder.DEFAULT_PARAMS.copy()
+        local_param['flexible_trip_lengths[]'] = "one_month"
+        local_param['flexible_trip_dates[]'] = month
+        return self._get_full_url(local_param)
+
+    def get_dates(self, check_in, check_out):
+        """
+        Generate a URL for a specific date range search.
+
+        Args:
+            check_in (str): Check-in date in 'YYYY-MM-DD' format.
+            check_out (str): Check-out date in 'YYYY-MM-DD' format.
+
+        Returns:
+            str: A URL for a specific date range search.
+        """
+        local_param = AirBnbURLBuilder.DEFAULT_PARAMS.copy()
+        local_param['checkin'] = check_in
+        local_param['checkout'] = check_out
+        local_param['flexible_date_search_filter_type'] = "1"
+        return self._get_full_url(local_param)
+
+    def get_months(self, check_in="2024-07-01", check_out="2024-10-29", length=3):
```

Comment:
would it be possible to get these auto-generated to a future date instead of hardcoded default params? Let's say this method gets called but it's August 2025 and no params are passed. I would believe the function will arbitrarily get these values but I'm not sure about the repercussions.

Reply:
Yes, you are correct. The default should be current date and three months in the future.